### PR TITLE
Test/97 wallet generated custom private key account

### PR DESCRIPTION
**Pull request by Valentina1133**
https://ethstatus.testrail.net/index.php?/cases/view/703033
https://ethstatus.testrail.net/index.php?/cases/view/703028
https://ethstatus.testrail.net/index.php?/cases/view/703029

Screenshot 2023-09-05 at 15 17 57

### CODE REVIEW

**Comment by anastasiyaig** [comment]
i launched new tests and they look okay-ish. I noticed that we have a problem when running tests in a batch (the previous process is active when we start using new one)

```
12:23:05.170   DEBUG driver.context  Attaching to AUT 'AUT' failed: another test is currently attached to the AUT.
12:23:06.177   DEBUG driver.context  Attaching to AUT 'AUT' failed: connection to AUT refused.
```

this is something that Vlad is working on and it will be addressed in another branch

**Review by anastasiyaig — APPROVED** [review]

File: test/ui-pytest/gui/elements/qt/object.py [hunk 1]
```diff
@@ -117,6 +117,11 @@ def _hover():
 
         assert driver.waitFor(lambda: _hover(), timeout_msec)
 
+    @allure.step('Hover and return')
```
Comment (StateOf-Vlado) [inline test/ui-pytest/gui/elements/qt/object.py:120]: If you need hover - use hover that we have If you need to return self in hover - add it, we don't need the method that executes the same things

File: test/ui-pytest/constants/wallet.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from enum import Enum
+
+
+class DerivationPath(Enum):
+    CUSTOM = 'Custom'
+    ETHEREUM = 'Ethereum'
+    ETHEREUM_ROPSTEN = 'Ethereum Testnet (Ropsten)'
+    ETHEREUM_LEDGER = 'Ethereum (Ledger)'
+    ETHEREUM_LEDGER_LIVE = 'Ethereum (Ledger Live/KeepKey)'
```
Comment (StateOf-Vlado) [inline test/ui-pytest/constants/wallet.py:9]: Please verify that you have one empty line at the end of all added files

Comment (StateOf-Vlado) [reply test/ui-pytest/constants/wallet.py:9]: image
As I understand GitHud shows this icon if you don't have it

File: test/ui-pytest/gui/components/authenticate_popup.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import allure
+
+from gui.elements.qt.button import Button
+from gui.elements.qt.text_edit import TextEdit
+from gui.elements.qt.object import QObject
+
+
+
+class AuthenticatePopup(QObject):
```
Comment (StateOf-Vlado) [inline test/ui-pytest/gui/components/authenticate_popup.py:9]: Two lines before classes
Please use hotkey to refactor your code:
https://www.jetbrains.com/help/pycharm/reformat-file-dialog.html#:~:text=The%20dialog%20appears%20when%20you,of%20the%20specified%20scope%20automatically.

File: test/ui-pytest/gui/components/authenticate_popup.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    def __init__(self):
+        super(AuthenticatePopup, self).__init__('contextMenu_PopupItem')
+        self._password_text_edit = TextEdit('sharedPopup_Password_Input')
+        self._primary_button = Button('sharedPopup_Primary_Button')
+
+    @allure.step('Authenticate in popup')
```
Comment (StateOf-Vlado) [inline test/ui-pytest/gui/components/authenticate_popup.py:16]: Please use business logic in steps, not what we are doing with UI but what we are doing in the current process:
Authenticate actions with password?

File: test/ui-pytest/gui/components/wallet/remove_wallet_account_popup.py [hunk 5]
```diff
@@ -22,4 +22,9 @@ def wait_until_appears(self, timeout_msec: int = configs.timeouts.UI_LOAD_TIMEOU
     @allure.step('Confirm removing account')
     def confirm(self):
         self._confirm_button.click()
-        self._confirm_button.wait_until_hidden()
\ No newline at end of file
+        self._confirm_button.wait_until_hidden()
+
+    @allure.step('Agree in have pen and paper checkbox and confirm removing account')
```
Comment (StateOf-Vlado) [inline test/ui-pytest/gui/components/wallet/remove_wallet_account_popup.py:27]: Agree and confirm removing account?

**Review by StateOf-Vlado — APPROVED** [review]
I left a few minor comments. 
- Do refactoring of your code every time before committing it. Pycharm can clean code: spaces, imports all will be fixed automatically
- remove hover_and_return, just add return self in hover method

Looks, good.

File: test/ui-pytest/gui/components/authenticate_popup.py (hunk 3 above)
Comment (Valentina1133) [reply test/ui-pytest/gui/components/authenticate_popup.py:8]: Fixed. And thank you @StateOf-Vlado, it's very useful

File: test/ui-pytest/gui/elements/qt/object.py (hunk 1 above)
Comment (Valentina1133) [reply test/ui-pytest/gui/elements/qt/object.py]: Fixed

File: test/ui-pytest/constants/wallet.py (hunk 2 above)
Comment (Valentina1133) [reply test/ui-pytest/constants/wallet.py]: Added

File: test/ui-pytest/gui/components/authenticate_popup.py (hunk 4 above)
Comment (Valentina1133) [reply test/ui-pytest/gui/components/authenticate_popup.py]: Fixed

File: test/ui-pytest/gui/components/wallet/remove_wallet_account_popup.py (hunk 5 above)
Comment (Valentina1133) [reply test/ui-pytest/gui/components/wallet/remove_wallet_account_popup.py]: Fixed

**Pull request merged**